src/rosdep2/rospkg_loader.py

```diff
@@ -78,6 +78,20 @@ def __init__(self, rospack=None, rosstack=None, underlay_key=None):
         self._loadable_resource_cache = None
         self._catkin_packages_cache = None
 
+        # Dependency types to include
+        def check_dep(dep_type):
+            if dependency_types:
+                return dep_type in dependency_types
+            else:
+                return dep_type in ['buildtool', 'build', 'build_export', 'exec', 'test']
+
+        self.include_build_depends = check_dep('build')
```

Comment:
this could probably be

```
# these should probably be constants defined in one place and used across this PR
all_dep_types = {'buildtool', 'build', 'build_export', 'exec', 'test', 'doc'}
default_dep_types = all_dep_types - {'doc'}

self.include_dep_types = all_dep_types.intersection(set(dependency_types)) if dependency_types else default_dep_types
```

then below in `get_rosdeps` we could use a generic iteration as well

```
deps = sum((getattr(pkg, f'{d}_depends') for d in self.include_dep_types), [])
```

In fact, after writing that, one way to get `all_dep_types` as above could be to introspect the `pkg` type object with something like

```
all_dep_types = [x for x in dir(pkg) if x.endswith('_depends')]
```

Reply:
I was unhappy with this floating function... Very nice suggestions! I'll consider applying them.

Reply:
Implemented this suggestion in fa3fd33. I still didn't find a good format for `all_dep_types`. For now, it is a function in rospkg_load.py.